Changed Upcoming Releases format. (#23020)
gravitational/teleport: docs/pages/preview/upcoming-releases.mdx

## Teleport

### Teleport 12.2.0

#### New hardened container images

New signed and hardened container images built on a distroless base which
results in a reduced attack surface and smaller footprint.

#### (Preview) IP-based validation for Kubernetes and Database Access

Teleport will be able to pin Kubernetes and database credentials to a
particular IP to prevent exfiltrated credentials from being used in a different
environment.

_Delayed from Teleport 12.1.0._

#### (Preview) Oracle support for Database Access

Teleport will add the ability to connect to Oracle databases to Teleport

> Database Access. Note, that Audit Log support will come in a later release.

Reviewer: ```suggestion
Database Access (Audit Log support will come in a later release).
```

| March 13th, 2023 | Teleport 12.1 begins to roll out for Cloud customers                                        |
| May 8th, 2023    | Add support for hosted Access Request plugins                                               |

### Production readiness

Teleport follows [semantic versioning](https://semver.org) for pre-releases and
releases.

#### Pre-releases

Pre-releases of Teleport (versions with suffixes like `-alpha`, `-beta`, `-rc`)
should not be run in production environments.

Pre-releases of Teleport are great for testing new features, breaking changes,
and backwards incompatibility issues either in development or staging
environments.

#### Major Releases

Major releases look like `(=teleport.major_version=).0.0`.

Major releases of Teleport contain many large new features and may contain
breaking changes.

Due to the scope and quantity of changes in a major release, we encourage
deploying to staging first to verify your usage pattern has not changed.

#### Minor Releases

Minor releases look like `(=teleport.major_version=).1.0`.

Minor releases of Teleport typically contain smaller features and improvements.

> Minor releases can be deployed directly to production.

Reviewer 2: Minor releases can typically be deployed directly to production.

The first minor release, like `(=teleport.major_version=).1.0`, is when most
customers upgrade to the next major version of Teleport.

#### Patch Releases

> Patch releases contain small bug fixes and are always ready for production use.

Reviewer 2: ```suggestion
Patch releases contain small bug fixes and can typically be deployed directly to production.
```
